src/python/foglamp/device/server.py

```diff
@@ -126,5 +131,40 @@ def start(cls, plugin, core_mgt_host, core_mgt_port):
                 signal_name,
                 lambda: asyncio.ensure_future(cls._stop(loop)))
 
+        microservice_management_app = web.Application(middlewares=[middleware.error_middleware])
+        routes.setup(microservice_management_app)
```

Comment:
and routes point to ? service registry?

Reply:
microservice_management_app should be device_management_app

Reply:
put into a try catch block? 

Move start and shutdown in two def? and start_app return a tuple?
Sort of:

```
serv_generater, handler, app = loop.run_until_complete(start_app(loop))
serv = loop.run_until_complete(serv_generator)
    log.debug('start server', serv.sockets[0].getsockname())
   
try:
   # register here; run that co-routine before loop starts
    loop.run_forever()
except KeyboardInterrupt:
    log.debug(' Stop server begin')
finally:
    loop.run_until_complete(shutdown(serv, app, handler))
    loop.close()
log.debug('Stop server end')
```

AND 

```
async def shutdown(server, app, handler):
    server.close()
    await server.wait_closed()

    app.client.close()  # database connection close
    await app.shutdown()
    await handler.finish_connections(10.0)
    await app.cleanup()
```

Reply:
I would say microservice_management_app is a better name as it is Maning the micro service elements and not managing the device elements. So I would say leave this as it is personally.

Reply:
> and routes point to ? service registry?
Yes. Currently the ping is also in service_registry, along with shutdown (unimplemented), etc.